### PR TITLE
feat: export controller function's FULLPATH (#131)

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

pick from https://github.com/eggjs/egg-core/pull/131

### CODE REVIEW

File: lib/loader/mixin/controller.js [hunk 1]
```diff
@@ -95,8 +97,9 @@ function wrapObject(obj, path) {
         throw new Error(`controller \`${key}\` should not use next as argument from file ${path}`);
       }
       ret[key] = functionToMiddleware(obj[key]);
+      ret[key][FULLPATH] = path + '#' + (prefix || '') + key + '()';
     } else if (is.object(obj[key])) {
-      ret[key] = wrapObject(obj[key], path);
+      ret[key] = wrapObject(obj[key], path, `${key}.`);
```
Comment (popomore) [inline lib/loader/mixin/controller.js]: 如果三层呢

Comment (fengmk2) [reply lib/loader/mixin/controller.js]: 呃，应该将 prefix 这里也处理。。。 bug

File: lib/loader/mixin/controller.js [hunk 2]
```diff
@@ -85,18 +87,19 @@ function wrapClass(Controller) {
...
       const names = utility.getParamNames(obj[key]);
       if (names[0] === 'next') {
-        throw new Error(`controller \`${key}\` should not use next as argument from file ${path}`);
+        throw new Error(`controller \`${prefix || ''}${key}\` should not use next as argument from file ${path}`);
       }
       ret[key] = functionToMiddleware(obj[key]);
+      ret[key][FULLPATH] = `${path}#${prefix || ''}${key}()`;
     } else if (is.object(obj[key])) {
-      ret[key] = wrapObject(obj[key], path);
+      ret[key] = wrapObject(obj[key], path, `${prefix || ''}${key}.`);
```
Comment (fengmk2) [inline lib/loader/mixin/controller.js]: @popomore fixed

**Comment by fengmk2** [comment]
3.x 也修复了 https://github.com/eggjs/egg-core/pull/133

**Comment by fengmk2** [comment]
4.2.0

**Pull request merged**